File: UI_React/src/app/services/randomFinanceRecordService.ts
import RandomDescriptionService from "./randomDescriptionService";

import { IFinanceRecord } from "../api";

const RandomFinanceRecordService = {
    generateRandomFinanceRecord(): IFinanceRecord {
        const randomAmount = Math.round(Math.random() * 200 - 100);
        const randomDescription = randomAmount > 0
          ? RandomDescriptionService.getRandomPositiveDescription()
          : RandomDescriptionService.getRandomNegativeDescription();

        const randomRecord: IFinanceRecord = {
          id: crypto.randomUUID(),
          date: new Date(),
          amount: randomAmount,
          description: randomDescription,
        };

        return randomRecord;
      }
}

export default RandomFinanceRecordService;

File: UI_Blazor_WebApp/UI_Blazor_WebApp/Services/RandomFinanceRecordService.cs
using UI_Blazor_WebApp.Models;

namespace UI_Blazor_WebApp.Services
{
    public class RandomFinanceRecordService
    {
        private RandomDescriptionService randomDescriptionService;

        public RandomFinanceRecordService(RandomDescriptionService randomDescriptionService)
        {
            this.randomDescriptionService = randomDescriptionService;
        }

        public FinanceRecordModel GenerateRandomFinanceRecord()
        {
            var randomAmount = new Random().Next(-100, 100);
            var randomDescription = randomAmount > 0 
                ? randomDescriptionService.GetRandomPositiveDescription()
                : randomDescriptionService.GetRandomNegativeDescription();

            var randomRecord = new FinanceRecordModel
            {
                Id = Guid.NewGuid(),
                Date = DateTime.UtcNow,
                Amount = randomAmount,
                Description = randomDescription
            };

            return randomRecord;
        }
    }
}
